txhelpers/swap.go

```diff
@@ -0,0 +1,125 @@
+package txhelpers
+
+import (
+	"encoding/hex"
+	"fmt"
+	"time"
+
+	"github.com/decred/dcrd/chaincfg/v2"
+	"github.com/decred/dcrd/dcrec"
+	"github.com/decred/dcrd/dcrutil/v2"
+	"github.com/decred/dcrd/txscript/v2"
+)
+
+// AtomicSwapContractPushes models the data pushes of an atomic swap contract.
+type AtomicSwapContractPushes struct {
+	ContractAddress   dcrutil.Address `json:"contract_address"`
+	RecipientAddress  dcrutil.Address `json:"recipient_address"`
+	RefundAddress     dcrutil.Address `json:"refund_address"`
+	Locktime          int64           `json:"locktime"`
+	FormattedLocktime string          `json:"formatted_locktime"`
+}
+
+// The redeem script contains an op_code that identifies the redeemer of a
+// contract, which is typically either the Initiator or the Participant.
+// AtomicSwapUnknownEntity is defined in case an unexpected op_code is found.
+const (
+	AtomicSwapUnknownEntity string = "Unknown"
+	AtomicSwapInitiator     string = "Initiator"
+	AtomicSwapParticipant   string = "Participant"
+)
+
+// ParseInputScriptForSwapContract checks if a tx input redeems a swap contract
+// and returns the contract that was redeemed, along with a string describing the
+// identity of the redeemer.
+// Returns an empty contract script and nil error if the provided script does not
+// redeem a contract. Returns a non-nil error if the script could not be parsed.
+func ParseInputScriptForSwapContract(inputScriptHex string, params *chaincfg.Params) ([]byte, string, error) {
```

Comment:
Defined `ParseInputScriptForSwapContract` and `ParseAtomicSwapContract` here to make them usable outside dcrdata.